packages/pigeon/lib/cpp_generator.dart

```diff
@@ -1355,20 +1345,16 @@ ${prefix}reply(EncodableValue(std::move(wrapped)));''';
             'const auto* $argName = std::get_if<${hostType.datatype}>(&$encodableArgName);');
       } else if (hostType.isEnum) {
         if (hostType.isNullable) {
+          final String valueVarName = '${argName}_value';
+          indent.writeln(
+              'const int64_t $valueVarName = $encodableArgName.IsNull() ? 0 : $encodableArgName.LongValue();');
           if (apiType == ApiType.flutter) {
-            indent.writeln('${hostType.datatype} $argName;');
+            indent.writeln(
+                'const auto* $argName = $encodableArgName.IsNull() ? nullptr : &(${hostType.datatype})$valueVarName;');
           } else {
-            indent.writeln('std::optional<${hostType.datatype}> $argName;');
+            indent.writeln(
+                'const std::optional<${hostType.datatype}> $argName = $encodableArgName.IsNull() ? std::optional<${hostType.datatype}>() : std::make_optional<${hostType.datatype}>(static_cast<${hostType.datatype}>(${argName}_value));');
```

Comment:
Can `std::optional<${hostType.datatype}>()` just be `std::nullopt`? I think that works in ternaries.

Reply:
It didn't work when I tried it in this context, but it's possible I had a different issue.